app/medInria/medFilteringWorkspace.cpp

```diff
@@ -97,67 +78,60 @@ void medFilteringWorkspace::setupViewContainerStack()
 
 void medFilteringWorkspace::changeToolBoxInput()
 {
-    if(d->filteringToolBox.isNull())
-        return;
-
-    if(!d->inputContainer->view())
+    if(!selectorToolBox().isNull())
     {
-        d->filteringToolBox->clear();
-        return;
-    }
-
-    medAbstractLayeredView *layeredView = dynamic_cast<medAbstractLayeredView *>(d->inputContainer->view());
-    if(!layeredView)
-    {
-        qWarning() << "Non layered view are not supported in filtering workspace yet.";
-        d->filteringToolBox->clear();
-        return;
+        if(!d->inputContainer->view() && (selectorToolBox()->data() != NULL))
+        {
+            selectorToolBox()->clear();
+        }
+        else
+        {
+            medAbstractLayeredView *layeredView = dynamic_cast<medAbstractLayeredView *>(d->inputContainer->view());
+            selectorToolBox()->onInputSelected(layeredView->layerData(layeredView->currentLayer()));
+        }
     }
-    d->filteringToolBox->onInputSelected(layeredView->layerData(layeredView->currentLayer()));
 }
 
 /**
  * @brief adds metadata to the output and emits a signal outputDataChanged(medAbstractData *)
  */
 void medFilteringWorkspace::onProcessSuccess()
 {
-    if(d->filteringToolBox.isNull())
-        return;
-
-    d->filterOutput = d->filteringToolBox->currentToolBox()->processOutput();
-    if ( !d->filterOutput )
-        return;
-
-    qDebug() << "d->filterOutput->identifier()" << d->filterOutput->identifier();
-
-    medAbstractData *inputData(d->filteringToolBox->data());
+    if(selectorToolBox()->data())
+    {
+        d->filterOutput = selectorToolBox()->currentToolBox()->processOutput();
+        if ( d->filterOutput )
+        {
+            medAbstractData *inputData(selectorToolBox()->data());
 
-    if (! d->filterOutput->hasMetaData(medMetaDataKeys::SeriesDescription.key()))
-      {
-        QString newSeriesDescription = inputData->metadata ( medMetaDataKeys::SeriesDescription.key() );
-        newSeriesDescription += " filtered";
-        d->filterOutput->addMetaData ( medMetaDataKeys::SeriesDescription.key(), newSeriesDescription );
-      }
+            if (! d->filterOutput->hasMetaData(medMetaDataKeys::SeriesDescription.key()))
+            {
+                QString newSeriesDescription = inputData->metadata ( medMetaDataKeys::SeriesDescription.key() );
+                newSeriesDescription += " filtered";
+                d->filterOutput->addMetaData ( medMetaDataKeys::SeriesDescription.key(), newSeriesDescription );
+            }
 
-    foreach ( QString metaData, inputData->metaDataList() )
-      if (!d->filterOutput->hasMetaData(metaData))
-        d->filterOutput->addMetaData ( metaData, inputData->metaDataValues ( metaData ) );
+            foreach ( QString metaData, inputData->metaDataList() )
+                if (!d->filterOutput->hasMetaData(metaData))
+                    d->filterOutput->addMetaData ( metaData, inputData->metaDataValues ( metaData ) );
```

Comment:
Yep, it was an automatic indentation, i did not change this code, i am going to add brackets.